src/main/java/com/cloudbees/jenkins/support/impl/CustomLogs.java

```diff
@@ -0,0 +1,217 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.FileContent;
+import com.cloudbees.jenkins.support.api.SupportContext;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.init.Terminator;
+import hudson.logging.LogRecorder;
+import hudson.model.PeriodicWork;
+import hudson.security.Permission;
+import hudson.util.CopyOnWriteList;
+import hudson.util.io.RewindableFileOutputStream;
+import hudson.util.io.RewindableRotatingFileOutputStream;
+import io.jenkins.lib.support_log_formatter.SupportLogFormatter;
+import jenkins.model.Jenkins;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.logging.StreamHandler;
+
+/**
+ * Custom Log recorders files from the master.
+ */
+@Extension(ordinal = 100.0)
+public class CustomLogs extends Component {
+
+    private static final int MAX_ROTATE_LOGS = Integer.getInteger(CustomLogs.class.getName() + ".MAX_ROTATE_LOGS", 9);
+    private final Map<String,LogRecorder> logRecorders = Jenkins.get().getLog().logRecorders;
```

Comment:
If this is loaded only once at the beginning, new recorded would not be taken into account?

Reply:
I have mostly moved the code and haven't touched that part. 
Testing this in debug, they are taken into account and apparently the code is holding a reference to the `LogRecorderManager#logRecorders`.